### PR TITLE
Support resourceQuery in context dependencies.

### LINKED ISSUE #5730: Context dependencies should support resource queries
**Do you want to request a *feature* or report a *bug*?**

Feature

**What is the current behavior?**

There is no way to pass a resource query when using context dependencies via `require.context`. In other words, `this.resourceQuery` will never be set inside a loader when using `require.context`.

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

`this.resourceQuery` should be set in a loader for something like the following:

```js
const ctx = require.context('images?foo=bar', false, /\.jpg$/);
ctx('./a.jpg');
```

I have a test case and a proposed fix in my fork here - https://github.com/anuraaga/webpack/pull/1
If the approach seems reasonable, I would be happy to move the PR to this repo for tweaks/submission, otherwise someone more familiar with the webpack design should probably take over as I just stumbled to that solution :)

**If this is a feature request, what is motivation or use case for changing the behavior?**

Some loaders (e.g. sharp-loader) use `this.resourceQuery` to enable certain behavior. This functionality cannot be used with context dependencies due to having no way to set a resource query. There doesn't seem to be any reason context dependencies should not support this feature as the only difference compared to a normal dependency is multiple files are resolved at the same time and placed in a map.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

All

### PR DESCRIPTION
**Pull request by anuraaga**
**What kind of change does this PR introduce?**

Adds support for resource queries in context dependencies.

**Did you add tests for your changes?**

Yes

**If relevant, link to documentation update:**

N/A

**Summary**

Some loaders require resource query for proper usage, and cannot be used at all with context dependencies.

**Does this PR introduce a breaking change?**

No, it only affects code that adds a resource query to a context, which currently does not work at all.

**Other information**

Fixes #5730

### CODE REVIEW

File: lib/ContextModuleFactory.js [hunk 1]
```diff
@@ -42,7 +42,8 @@ module.exports = class ContextModuleFactory extends Tapable {
 			const asyncContext = result.async;
 			const dependencies = result.dependencies;
 
-			let loaders, resource, loadersPrefix = "";
+			let loaders, resource, loadersPrefix = "",
```
Comment (sokra) [inline lib/ContextModuleFactory.js]: split this into multiple `let` declarations.

**Comment by anuraaga** [comment]
@sokra Just wondering, when would this be merged into the next branch? It seems it needs a rebase, but would hold off if it'll just get out of date again

**Comment by anuraaga** [comment]
@sokra Ping?

**Comment by anuraaga** [comment]
@sokra I have gone ahead and rebased this PR - do you think it can be merged? It would be depressing for it to get out of sync with the branch again...

**Comment by sokra** [comment]
Thanks

**Pull request merged**